lib-multisrc/natsuid/src/eu/kanade/tachiyomi/multisrc/natsuid/NatsuId.kt

```
@@ -0,0 +1,378 @@
+                    Filter.Separator(),
+                    Filter.Header("Press 'reset' to load genre filter"),
+                ),
+            )
+
+            return@runBlocking FilterList(filters)
+        }
+
+        val responseString = response.body.string()
+        response.close()
+
+        val data = try {
+            responseString.cleanJsonResponse().parseAs<List<Term>>()
+        } catch (e: Throwable) {
+            Log.e(name, "Failed to parse genre filters", e)
+
+            filters.addAll(
+                listOf(
+                    Filter.Separator(),
+                    Filter.Header("Failed to parse genre filter"),
+                ),
+            )
+
+            return@runBlocking FilterList(filters)
+        }
+
+        filters.addAll(
+            listOf(
+                GenreFilter(
+                    data.map { it.name to it.slug },
+                ),
+                GenreInclusion(),
+                GenreInclusion(),
+            ),
+        )
+
+        FilterList(filters)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val document = Jsoup.parseBodyFragment(response.body.string(), baseUrl)
+        val slugs = document.select("div > a[href*=/manga/]:has(> img)").map {
+            it.absUrl("href").toHttpUrl().pathSegments[1]
+        }.ifEmpty {
+            return MangasPage(emptyList(), false)
+        }
+
+        val url = "$baseUrl/wp-json/wp/v2/manga".toHttpUrl().newBuilder().apply {
+            slugs.forEach { slug ->
+                addQueryParameter("slug[]", slug)
+            }
+            addQueryParameter("per_page", "${slugs.size + 1}")
+            addQueryParameter("_embed", null)
+        }.build()
+
+        val jsonResponse = client.newCall(GET(url, headers)).execute()
+        val cleanJson = jsonResponse.body.string().cleanJsonResponse()
+
+        val details = try {
+            cleanJson.parseAs<List<Manga>>()
+        } catch (e: Exception) {
```

don't swallow exceptions